be/src/olap/comparison_predicate.h

```diff
@@ -524,7 +526,7 @@ class ComparisonPredicateBase : public ColumnPredicate {
                 _base_loop_bit<is_nullable, is_and>(sel, size, flags, null_map, data_array,
                                                     dict_code);
             } else {
-                LOG(FATAL) << "column_dictionary must use StringRef predicate.";
+                static_assert(!std::is_same_v<T, StringRef>, "column_dictionary must use StringRef predicate.");
```

Comment:
we wrote `if constexpr (std::is_same_v<T, StringRef>)` upon
so here going into `else` means `T` is **not same** with `StringRef`. we want a failure here. so I think we should assert `std::is_same_v<T, StringRef>`, right?

Reply:
sorry I made a mistake here, and I have done fixing three similar errors in this file right away